File: angularapp/src/app/shared/contracts/endpoints/ItemEndpoints.ts
import { baseUrl } from '../../types';

const itemBaseUrl = `${baseUrl}/item`;

export const ItemEndpoints: Readonly<{
  create: string;
  search: string;
  myItems: string;
  inactive: string;
  active: string;
  bidden: string;
  getById: (id: number) => string;
  getImage: (guid: string) => string;
  categories: (id: number) => string;
  edit: (id: number) => string;
  activate: (id: number) => string;
  delete: (id: number) => string;
}> = {
  create: `${itemBaseUrl}`,
  search: `${itemBaseUrl}`,
  myItems: `${itemBaseUrl}/myitems`,
  inactive: `${itemBaseUrl}/inactive`,
  active: `${itemBaseUrl}/active`,
  bidden: `${itemBaseUrl}/bidden`,
  getById: (id: number) => `${itemBaseUrl}/${id}`,
  getImage: (guid: string) => `${itemBaseUrl}/${guid}`,
  categories: (id: number) => `${itemBaseUrl}/categories/${id}`,
  edit: (id: number) => `${itemBaseUrl}/edit/${id}`,
  activate: (id: number) => `${itemBaseUrl}/activate/${id}`,
  delete: (id: number) => `${itemBaseUrl}/${id}`,
};

File: Contracts/Endpoints/ItemEndpoints.cs
﻿namespace contracts.Endpoints
{
    public static class ItemEndpoints
    {
        public const string BaseUrl = "api/Item";

        public const string Create = "";

        public const string Search = "";

        public const string GetImage = "{guid}";

        public const string Categories = "Categories/{id:long}";

        public const string Inactive = "Inactive";

        public const string Active = "Active";

        public const string Bidden = "Bidden";

        public const string GetById = "{id:long}";

        public const string Activate = "Activate/{id:long}";

        public const string Edit = "Edit/{id:long}";

        public const string Delete = "{id:long}";
    }
}
